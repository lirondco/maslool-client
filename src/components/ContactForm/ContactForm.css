.contact_form p {
  margin-top: 6%;
  margin-left: 10%;
}

.contact_form p span {
  color: red;
  font-weight: bolder;
}

.contact_form form {
  margin-left: 10% !important;
}

.contact_form form label {
  margin-left: -15% !important;
}
