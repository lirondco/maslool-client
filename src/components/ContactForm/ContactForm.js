import React, { Component } from "react";
import UserContext from "../../contexts/UserContext";
import PendingApiService from "../../services/pending-api-service";
import { Button, Label, Required, Textarea } from "../Utils/Utils";
import "./ContactForm.css";

export default class ContactForm extends Component {
  state = {
    submitted: false,
  };

  static contextType = UserContext;

  handleSubmit = (ev) => {
    this.context.clearError();
    ev.preventDefault();
    const { message } = ev.target;
    PendingApiService.postPending(message.value)
      .then((res) => this.setState({ submitted: true, ...res }))
      .then(() => {
        message.value = "";
      })
      .catch(this.context.setError);
  };

  renderForm = () => {
    const { user, error } = this.context;
    return (
      <>
        <p>
          From: <span className="username_contact">{user.username}</span>
        </p>
        <form
          type="submit"
          className="contact_form"
          onSubmit={this.handleSubmit}
        >
          <Label htmlFor="contact_form_message">
            Write your message: <Required />
          </Label>
          <Textarea required id="contact_form_message" name="message" />
          <Button>Submit Message</Button>
          <div role="alert">{error && <p className="error">{error.error}</p>}</div>
        </form>
      </>
    );
  };

  renderSent = () => {
    const { user } = this.context;
    return (
      <>
        <p>
          Thank you, <span className="username_contact">{user.username}</span>.
          Your message was sent and we will contact your email that we have on
          file. Here's your message:
        </p>
        <div className="contact_form">
          <p className="contact_form_message">{this.state.message}</p>
        </div>
      </>
    );
  };

  render() {
    return (
      <div className="contact_form">
        {!this.state.submitted ? this.renderForm() : this.renderSent()}
      </div>
    );
  }
}
