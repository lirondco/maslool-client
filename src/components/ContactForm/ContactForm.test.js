import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ContactForm from "./ContactForm";
import { withContext } from "shallow-with-context";

describe(`ContactForm component`, () => {
  const context = {
    clearError: () => {},
    user: { id: 3, admin: false, banned: false, username: "test_user" },
  };
  const ComponentWithContext = withContext(ContactForm, context);

  it("renders the form", () => {
    const wrapper = shallow(<ComponentWithContext />, { context });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("form submits when submitted", () => {
    const wrapper = shallow(<ComponentWithContext />, { context });
    const mockedEvent = {
      preventDefault: () => {},
      target: { message: { value: "Test" } },
    };
    wrapper.find("form").at(0).simulate("submit", mockedEvent);
  });
});
