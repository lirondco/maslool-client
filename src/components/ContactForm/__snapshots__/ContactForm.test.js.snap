// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component renders the form 1`] = `
<div
  className="contact_form"
>
  <p>
    From: 
    <span
      className="username_contact"
    >
      test_user
    </span>
  </p>
  <form
    className="contact_form"
    onSubmit={[Function]}
    type="submit"
  >
    <Label
      htmlFor="contact_form_message"
    >
      Write your message: 
      <Required />
    </Label>
    <Textarea
      id="contact_form_message"
      name="message"
      required={true}
    />
    <ForwardRef>
      Submit Message
    </ForwardRef>
    <div
      role="alert"
    />
  </form>
</div>
`;
